src/AutoRest.CSharp/LowLevel/Decorator/MethodAmbiguityResolver.cs

```diff
@@ -0,0 +1,169 @@
+                    }
+                    ambiguityFound = true;
+                }
+            };
+
+            if (ambiguityFound)
+            {
+                return clientMethod.CloneWithAllParametersRequired();
+            }
+            return clientMethod;
+        }
+
+        /// <summary>
+        /// Check if a client method is ambiguous with the method in source model.
+        /// Two methods are considered as ambiguous when:
+        /// - both has the same number of required parameters
+        /// - for each required parameter, either they should be the same type in source model.
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="methodSymbol"></param>
+        /// <returns></returns>
+        private static bool IsAmbiguousMethod(LowLevelClientMethod clientMethod, IMethodSymbol methodSymbol)
+        {
+            var clientMethodRequiredParameters = clientMethod.Signature.Parameters.Where(p => !p.IsOptionalInSignature).ToList();
+            var requiredParameterSymbols = methodSymbol.Parameters.Where(p => !p.HasExplicitDefaultValue).ToList();
+            if (clientMethodRequiredParameters.Count != requiredParameterSymbols.Count)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < clientMethodRequiredParameters.Count; i++)
+            {
+                if (!IsSameParameter(clientMethodRequiredParameters[i], requiredParameterSymbols[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Check if a clientMethod has the same paramters as a method from source model has.
+        /// Both methods should:
+        /// - have the same number of parameters
+        /// - each parameter has the same type
+        /// - each parameter has the same default value (e.g. optional or required)
+        /// </summary>
+        /// <param name="clientMethod"></param>
+        /// <param name="clientMethodSourceModel"></param>
+        /// <returns></returns>
+        private static bool IsSameMethod(LowLevelClientMethod clientMethod, IMethodSymbol clientMethodSourceModel)
+        {
+            if (clientMethod.Signature.Parameters.Count != clientMethodSourceModel.Parameters.Length)
+            {
+                return false;
+            }
+            for (int i = 0; i < clientMethod.Signature.Parameters.Count; i++)
+            {
+                if (!IsSameParameter(clientMethod.Signature.Parameters[i], clientMethodSourceModel.Parameters[i]))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool IsSameParameter(Parameter parameter, IParameterSymbol parameterSourceModel)
+        {
+            //if (parameter.Name != parameterSourceModel.Name)
+            //{
+            //    return false;
+            //}
```

Comment:
I don't think we need to consider name either.

Reply:
Ah, yes, I forgot to delete the commented coede.

Reply:
Does the parameter name part of public API? Because our customer could always use the "named parameter pattern" even if the parameter is not optional, like this:
```csharp
var resp = client.CreateSomething(requestContent: requestContent, soemthingElse: something);
```

Reply:
@ArcturusZhang In this case, we plan to make sure the generated new method is different from the existing one. So if the new method breaks the codes, it's actually a good thing since the code above will invoke the old method.